client/pages/offer-thread.jsx

```diff
@@ -17,65 +22,121 @@ class OfferThread extends React.Component {
     })
       .then(response => response.json())
       .then(offerThread => {
-        this.setState({ message: offerThread[0] });
+        this.setState({ messageIn: offerThread });
       });
+    const socket = io.connect('/', {
+      transports: ['websocket'],
+      reconnectionDelayMax: 1000,
+      query: {
+        postId: this.props.postId,
+        userToken: window.localStorage.getItem('user-jwt'),
+        recipientId: window.localStorage.getItem('userId')
+      }
+    });
+    socket.on('connect', () => {
+      // eslint-disable-next-line no-console
+      console.log('client connected');
+    });
+  }
+
+  handleMessage(event) {
+    this.setState({ reply: event.target.value });
+  }
+
+  handleSubmit(event) {
+    const token = window.localStorage.getItem('user-jwt');
+    const replyObj = {
+      reply: this.state.reply,
+      recipient: this.props.senderId,
+      postId: this.props.postId
+    };
+
+    event.preventDefault();
+    fetch('/api/messages', {
+      method: 'POST',
+      headers: {
+        'content-type': 'application/json',
+        'X-Access-Token': token
+      },
+      body: JSON.stringify(replyObj)
+    })
+      .then(response => response.json())
+      .then(reply => {
+
+      })
+      .catch(err => console.error(err));
+    this.setState({ reply: '' });
   }
 
   render() {
-    if (!this.state.message) return null;
-    const { message, imgUrl, title, username, price, postId } = this.state.message;
+    if (!this.state.messageIn) return null;
+    const { username } = this.state.messageIn.user;
+    const postId = this.state.messageIn.postId;
+    const title = this.state.messageIn.title;
+    const price = this.state.messageIn.price;
+    const imgUrl = this.state.messageIn.imgUrl;
+
     return (
       <div className='container'>
-        <div className='modal-row'>
-          <div className='col-half ml-desktop mt-desktop-2rem '>
-            <h3 className='roboto-4 display-inline pdl-mobile'>{username}</h3>
-            <div className='border-bottom-2px'></div>
-            <div className='modal-row height-align-bottom'>
-              <div className='column-100'>
-                  <div className="chat">
-                    <div className="yours messages">
-                      <div className="message in last">
-                        <p className='roboto-4'>{message}</p>
+
+                <div className='modal-row'>
+                  <div className='col-half ml-desktop mt-desktop-2rem '>
+                    <h3 className='roboto-4 display-inline pdl-mobile'>{username}</h3>
+                    <div className='border-bottom-2px'></div>
+                    <div className='modal-row height-align-bottom'>
+                      <div className='column-100'>
+                <div className="chat" >
+                  {this.state.messageIn.message.map(
+                    (message, index) => {
+                      const sellerId = Number(window.localStorage.getItem('userId'));
```

Comment:
It's not good to access localstorage in render, it's synchronous (blocking) and you don't want to do it in `map()` because the same value is being read over and over again. instead, pass the current user's id into the component from App or something.